strawberry/relay/fields.py

```diff
@@ -253,13 +256,18 @@ def apply(self, field: StrawberryField) -> None:
                 None,
             )
 
+        if is_union(resolver_type):
+            # TODO: actually check if is optional and get correct type
+            resolver_type = get_args(resolver_type)[0]
```

Comment:
**suggestion:** we can probably use `is_optional` to check and `get_optional_annotation` in this line from `strawberry.utils.typing`